### PR TITLE
Account invoice list and filters

### PR DESCRIPTION
**Pull request by nmsan**

### CODE REVIEW

File: database/seeders/AccountInvoicesSeeder.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+{
+    /**
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        //
+        $this-> call(QuotationSeeder::class);
```
Comment (nuke99) [inline database/seeders/AccountInvoicesSeeder.php:20]: remove `$this-> call(QuotationSeeder::class);` and the `AccountInvoicesSeeder` class to the main database seeder

**Review by nuke99 — CHANGES_REQUESTED** [review]

File: database/seeders/AccountInvoicesSeeder.php (hunk 1 above)
Comment (nmsan) [reply database/seeders/AccountInvoicesSeeder.php]: done

File: app/Http/Resources/Invoice/AccountInvoiceResource.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
+<?php
+
+namespace App\Http\Resources\Invoice;
+
+use Illuminate\Http\Resources\Json\JsonResource;
+
+class AccountInvoiceResource extends JsonResource
```
[the remark this answers is not in the document]
Comment (nmsan) [reply app/Http/Resources/Invoice/AccountInvoiceResource.php:7]: Now using a collection to solve pagination issue

File: app/Http/Controllers/AccountInvoiceController.php [hunk 3]
```diff
@@ -19,9 +22,19 @@
...
-        return Inertia::render('Accounts/Invoices/Index');
+        $invoices = Invoice::query()
+            ->when($request->filled('invoice_number'), function ($query) use ($request) {
+                return $query->where('invoice_number', 'LIKE', '%'.$request->get('invoice_number').'%');
+            })
+            ->paginate(2)
+            ->withQueryString()
+        ;
+        return Inertia::render('Accounts/Invoices/Index', [
+            'invoices' => new AccountInvoiceCollection($invoices),
```
Comment (nuke99) [inline app/Http/Controllers/AccountInvoiceController.php:35]: we can also do this by `AccountInvoiceResource::collection($invoices)`

File: app/Http/Controllers/AccountInvoiceController.php [hunk 4]
```diff
@@ -9,19 +9,30 @@
...
-        return Inertia::render('Accounts/Invoices/Index');
+        $invoices = Invoice::query()
+            ->with('quotation')
+            ->when($request->filled('invoice_number'), function ($query) use ($request) {
+                return $query->where('invoice_number', 'LIKE', '%'.$request->get('invoice_number').'%');
+            })
+            ->paginate()
+            ->withQueryString()
+        ;
+//        dd($invoices);
```
Comment (nuke99) [inline app/Http/Controllers/AccountInvoiceController.php:31]: remove this

File: database/seeders/AccountInvoicesSeeder.php [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+     * @return void
+     */
+    public function run()
+    {
+        //
+        Invoice::create([
+            'aggregate_id' => Str::uuid()->toString(),
+            'xero_invoice_id' => 'xero-1',
+            'invoice_number' => 'invoice-1',
+            'quotation_id' => 1,
```
Comment (nuke99) [inline database/seeders/AccountInvoicesSeeder.php:23]: Fetch this from static method from QuotationSeeder

**Review by nuke99 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/AccountInvoiceController.php (hunk 3 above)
Comment (nmsan) [reply app/Http/Controllers/AccountInvoiceController.php]: changed

File: app/Http/Controllers/AccountInvoiceController.php (hunk 4 above)
Comment (nmsan) [reply app/Http/Controllers/AccountInvoiceController.php]: removed

File: database/seeders/AccountInvoicesSeeder.php (hunk 5 above)
Comment (nmsan) [reply database/seeders/AccountInvoicesSeeder.php]: changed

**Pull request merged**